app/controllers/products_controller.rb

```diff
@@ -32,4 +32,17 @@ def update
       end
     end
   end
+
+  def vote
+    product = Product.find(params[:product_id])
+    vote = Vote.where(:user_id => current_user.id, :product_id => product.id).first
+     if vote.present?
+       vote.destroy
+     else
+      Vote.create(:user_id => current_user.id, :product_id => product.id)    
+     end
+    votes = product.votes.count
+    render json: votes.as_json(methods: :votes)
```

Comment:
Lets not return all the votes records, really not necessary to send all this over the wire. All this can be aggregated.

We should be sending a contextual true or false based on a user.

    render json: product.as_json(methods: :votes_count).merge("hasVoteForUser" => product.has_vote_for(current_user))

By sending back the product instance we are sticking to REST, and also get the new vote count for free

This would require 2 new methods on `Product`, `#votes_count` and `#has_vote_for(user)`.